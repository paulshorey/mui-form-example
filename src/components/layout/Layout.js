//@flow
import React, { Component } from 'react';
import { connect } from 'react-redux';
import * as uiActions from 'redux/actions/ui'; 

import Top from 'components/layout/components/Top';
import Bottom from 'components/layout/components/Bottom';
import Nav from 'components/nav/Nav';
import Message from 'components/layout/components/Message';
import Side from 'components/layout/components/Side';
import Popup from 'components/layout/components/Popup';
import SpecificErrorBoundary from 'components/error/SpecificErrorBoundary';

import * as Styled from './LayoutStyled';

type Props = {
	ui: {}
};
class Layout extends Component<Props> {
	render() {
		var { history } = this.props;
		return (
			<Styled.Layout>
				<SpecificErrorBoundary location="Layout.js">
					<Message history={history} />
					<div
						className={'layout_container' + (this.props.routes ? ' withNav' : ' noNav')}
						tabIndex={0}
						onKeyDown={e => {
							if (e.keyCode === 27) {
								/*this.props.dispatch( layoutActions.popupBox({}) );*/
							}
						}}
					>
						<Top history={history} />

						<div className="layout_content">
							{this.props.routes && (
								<Nav history={history} routes={this.props.routes} />
							)}

							<div
								className="page"
								onClick={() => {
									this.props.dispatch(uiActions.UI_NAV_CLOSE());
								}}
							>
								{this.props.children}
							</div>
						</div>

						<Bottom history={history} routes={this.props.routes} />

						<Side history={history} />
						<Popup history={history} />
					</div>
				</SpecificErrorBoundary>
			</Styled.Layout>
		);
	}
}

const mapStateToProps = (state) => {
	return {
		ui: state.ui || {}
  	}
}

export default connect(mapStateToProps)(Layout);

