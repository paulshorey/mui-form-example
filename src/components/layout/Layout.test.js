import React from 'react';
import renderer from 'react-test-renderer';

// shallow + redux
import { shallow, connected, shallowConnected } from 'index.testing';
import { Layout } from './Layout';

it('renders correctly', () => {
	const tree = renderer
		.create(
			shallowConnected(
				<Layout>
					<div />
				</Layout>
			)
		)
		.toJSON();
	expect(tree).toMatchSnapshot();
});
