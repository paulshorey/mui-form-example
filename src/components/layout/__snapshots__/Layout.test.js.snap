// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="sc-iwsKbI gnbXsY"
>
  <div
    className="sc-gzVnrw bYkUlK"
  >
    <div
      className="messages_container"
    >
      <div
        className="messages_content"
      >
        <div
          className="message"
          onMouseEnter={[Function]}
        >
          <div
            className="message_content"
          >
            <div
              className="messageX"
              onClick={[Function]}
            >
              <span
                className="fontIcon icon-cross"
                title="hide message"
              />
            </div>
            <div
              className="message_text messageTitle"
            >
              <div>
                <span>
                   
                </span>
                 
                
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="layout_container noNav"
    onKeyDown={[Function]}
    tabIndex={0}
  >
    <div
      className="sc-bwzfXH Xhbhr"
    >
      <div>
        <div
          className="sc-ifAKCX bIPvnp"
          id="Hamburger"
          onClick={[Function]}
        >
          <div
            className="hamburger-icon"
          >
            <div
              className="hamburger-centered"
            >
              <div
                className="hamburger-top"
              />
              <div
                className="hamburger-middle"
              />
              <div
                className="hamburger-bottom"
              />
            </div>
          </div>
        </div>
        <div
          className="sc-htpNat ipbLAt"
        >
          <a
            href="/"
          >
            <img
              alt="LUXUL"
              src="/branding/logo/logo_white.svg"
            />
          </a>
        </div>
        <div
          className="sc-bxivhb kLTdtH"
          onClick={[Function]}
        >
          <span
            className="link"
          >
            <span
              className="title"
              style={
                Object {
                  "paddingTop": "0.1rem",
                  "textIndent": "-1.5px",
                }
              }
            >
              
            </span>
            <span
              className="text"
              style={
                Object {
                  "textIndent": "-1px",
                }
              }
            >
              Firmware: 
              
            </span>
          </span>
          <span
            className="fontIcon icon-navlink_info"
          />
        </div>
      </div>
    </div>
    <div
      className="layout_content"
    >
      <div
        className="page"
        onClick={[Function]}
      >
        <div />
      </div>
    </div>
    <div
      className="Bottom sc-EHOje fNRtIu"
    >
      <span
        className="logo"
      >
        <img
          alt="Legrand"
          src="/branding/logo/legrand_white.svg"
        />
      </span>
      <span
        className="copyright"
      >
        <span
          className="desktop_only"
        >
          This app is non-functional without the Mui hardware. 
        </span>
        <a
          className="color_success"
          href="/XWC1001/examples/validation"
          style={
            Object {
              "fontWeight": "bold",
            }
          }
        >
          CLICK HERE TO DEMO the UI »
        </a>
      </span>
    </div>
    <div
      className="sideBox  sc-htoDjs dXTkss"
    >
      <div
        className="sideBox_overlay"
        onClick={[Function]}
      />
      <div
        className="sideBox_content"
      >
        <div
          className="sideBox_close"
          onClick={[Function]}
        >
          <span
            className="fontIcon icon-cross"
            title="Close"
          />
        </div>
      </div>
    </div>
    <div
      className="popupBox  sc-dnqmqq feMKQw"
    >
      <div
        className="popupBox_overlay"
        onClick={[Function]}
      />
      <div
        className="popupBox_content"
      >
        <div
          className="popupBox_close"
          onClick={[Function]}
        >
          <span
            className="fontIcon icon-cross"
            title="Close"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
